recipes/darwin-toolchain/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+import os
+import platform
+import copy
+
+
+class DarwinToolchainConan(ConanFile):
+    name = "darwin-toolchain"
+    version = "1.0.9"
+    license = "Apple"
+    settings = "os", "arch", "build_type"
+    options = {"bitcode": [True, False]}
+    default_options = {"bitcode": True}
+    description = "Darwin toolchain to (cross) compile for macOS/iOS/watchOS/tvOS"
+    homepage = "https://developer.apple.com"
+    topics = ("xcode", "bitcode", "darwin")
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = "darwin-toolchain.cmake"
+
+    @property
+    def cmake_system_name(self):
+        if self.settings.os == "Macos":
+            return "Darwin"
+        return str(self.settings.os)
+
+    @property
+    def cmake_system_processor(self):
+        return {"x86": "i386",
+                "x86_64": "x86_64",
+                "armv7": "arm",
+                "armv8": "aarch64"}.get(str(self.settings.arch))
+
+    def config_options(self):
+        # build_type is only useful for bitcode
+        if self.settings.os == "Macos":
+            del self.settings.build_type
+            del self.options.bitcode
+
+    def validate(self):
+        if not tools.is_apple_os(self.settings.os):
+            raise ConanInvalidConfiguration("os must be an Apple os")
+        if self.settings.os in ["watchOS", "tvOS"] and not self.options.bitcode:
+            raise ConanInvalidConfiguration("bitcode is required on watchOS/tvOS")
+        if self.settings.os == "watchOS" and self.settings.arch not in ["armv7k", "armv8", "x86", "x86_64"]:
+            raise ConanInvalidConfiguration("watchOS: Only supported archs: [armv7k, armv8, x86, x86_64]")
+
+    def package(self):
+        self.copy("darwin-toolchain.cmake")
+
+    def package_info(self):
+        darwin_arch = tools.to_apple_arch(self.settings.arch)
+
+        if self.settings.os == "watchOS" and self.settings.arch == "armv8":
+            darwin_arch = "arm64_32"
+
+        sdk = None
+        if self.settings.os == "iOS":
+            if self.settings.get_safe("os.sdk") == "iPhoneSimulator":
+                sdk = "iphonesimulator"
+            else:
+                sdk = "iphoneos"
+        xcrun = tools.XCRun(self.settings, sdk)
+        sysroot = xcrun.sdk_path
+
+        self.cpp_info.sysroot = sysroot
```

Comment:
I've never seen this before. I tried to find the docs for it and I've only found (old) issues about it not being documented.
https://github.com/conan-io/docs/issues/703
https://github.com/conan-io/docs/issues/1343

Reply:
To be honest I don't remember why I needed it, but it is likely related to non-CMake generators.